lib/Sema/SemaBounds.cpp

```diff
@@ -711,14 +764,17 @@ namespace {
       // Bounds of the right-hand side of the assignment
       BoundsExpr *RHSBounds = nullptr;
 
-      // Check that the RHS of the assignment has bounds if the
+      // Check that the value being assigned has bounds if the
       // target of the LHS lvalue has bounds.
       if (LHSType->isCheckedPointerType() ||
-          LHSType->isIntegralOrEnumerationType()) {
+          LHSType->isIntegerType()) {
```

Comment:
I read the clang documentation and it says that isIntegralOrEnumerationType() is meant to be used for C++ code.  For C code, you can still with isIntegerType(), which treats enumerations as integers too.